File: src/main/java/org/jukeboxmc/block/BlockSculkCatalyst.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemSculkCatalyst;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockSculkCatalyst extends Block {

    public BlockSculkCatalyst() {
        super( "minecraft:sculk_catalyst" );
    }

    @Override
    public ItemSculkCatalyst toItem() {
        return new ItemSculkCatalyst();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.SCULK_CATALYST;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockSculkCatalyst.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.SculkCatalyst
import org.jukeboxmc.api.block.data.BlockFace
import org.jukeboxmc.api.blockentity.BlockEntity
import org.jukeboxmc.api.blockentity.BlockEntityType
import org.jukeboxmc.api.math.Vector
import org.jukeboxmc.server.block.JukeboxBlock
import org.jukeboxmc.server.extensions.toByte
import org.jukeboxmc.server.item.JukeboxItem
import org.jukeboxmc.server.player.JukeboxPlayer
import org.jukeboxmc.server.world.JukeboxWorld

class BlockSculkCatalyst(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates),
    SculkCatalyst {

    override fun placeBlock(
        player: JukeboxPlayer,
        world: JukeboxWorld,
        blockPosition: Vector,
        placePosition: Vector,
        clickedPosition: Vector,
        itemInHand: JukeboxItem,
        blockFace: BlockFace
    ): Boolean {
        BlockEntity.create(BlockEntityType.SCULKCATALYST, this)?.spawn()
        return super.placeBlock(player, world, blockPosition, placePosition, clickedPosition, itemInHand, blockFace)
    }

    override fun isBloom(): Boolean {
        return this.getBooleanState("bloom")
    }

    override fun setBloom(value: Boolean): SculkCatalyst {
        return this.setState("bloom", value.toByte())
    }
}